pkg/utils/validator/clusterresourceplacement.go

```diff
@@ -213,6 +223,30 @@ func validateClusterAffinity(clusterAffinity *placementv1beta1.ClusterAffinity,
 	return apiErrors.NewAggregate(allErr)
 }
 
+func validateTolerations(tolerations []placementv1beta1.Toleration) error {
+	tolerationMap := make(map[string]placementv1beta1.Toleration)
+	for _, toleration := range tolerations {
+		if toleration.Key == "" && toleration.Operator != corev1.TolerationOpExists {
+			return fmt.Errorf(invalidTolerationErrFmt, toleration)
+		}
+		if toleration.Value == "" && toleration.Operator == corev1.TolerationOpEqual {
+			return fmt.Errorf(invalidTolerationErrFmt, toleration)
+		}
+		key := buildTolerationKey(toleration)
+		_, exists := tolerationMap[key]
```

Comment:
IIRC, we only need to know if the key exists or not, why store the toleration as the value?